rappasoft/laravel-livewire-tables
# setFilterDefaultValue, Theming Optimization, TypeHint (PHP7.4) Fixes

### All Submissions:

### Changed
- General
  - Add capability for passing a custom model path to the MakeCommand
  - Add setFilterDefaultValue() on a per-component basis
  - Add getFilterDefaultValue() to each Filter Component (to maintain support for PHP7.4 - variable return types)
  - Moved Setting of Filter Defaults to Traits/Helpers/FilterHelpers - mountFilterHelpers
  - Moved Setting of Theme to Traits/ComponentUtilities - mountComponentUtilities for efficiency
  - Fix for TypeHint to allow continued support of PHP 7.4
  
- Filters
  - *Fix* - Changed the booting order to prevent repeated calling of filters() - https://github.com/rappasoft/laravel-livewire-tables/pull/1166 
  - *Fix* - fixed multiSelectDropdownFilter in menus - https://github.com/rappasoft/laravel-livewire-tables/pull/1160 

- Documentation
  - *Fix* - Minor wording tweak to documentation - https://github.com/rappasoft/laravel-livewire-tables/pull/1139 
  - *Fix* - Fix to example in applying-filters

- Tests
  - Restore PHP 7.4 tests for L8 only (includes minor tweaks)
  - Remove duplicate tests in ComponentHelpersTest
  - Tweak to ReorderingVisualsTest
  - Add TextFilter as a test in FilterHelpersTest



### New Feature Submissions:

1. [X] Does your submission pass tests and did you add any new tests needed for your feature?
2. [X] Did you update all templates (if applicable)?
3. [X] Did you add the [relevant documentation](https://github.com/rappasoft/laravel-livewire-tables-docs) (if applicable)?
4. Did you test locally to make sure your feature works as intended?

### Changes to Core Features:
